root/usr/share/fdi/facts/disks_partition_table.rb

```diff
@@ -0,0 +1,57 @@
+#
+# $ FACTERLIB=/usr/share/fdi/facts tfm-ruby /opt/theforeman/tfm/root/usr/bin/facter
+#
+
+require 'facter'
+
+class PartedInfo
+
+  attr_accessor :units, :size, :transport_type, :logical_sector_size, :physical_sector_size, :partition_table_type, :model_name, :flags
+
+  def initialize(disk)
+    parted_lines = `parted --machine --script /dev/#{disk} print 2>/dev/null`.split("\n", -1)
+    # https://git.savannah.gnu.org/cgit/parted.git/tree/parted/parted.c?h=v3.5#n1116
+    if parted_lines.length >= 2 &&
+      /^(?<units>CHS|CYL|BYT);$/ =~ parted_lines[0] &&
+      /^\/.+?:(?<size>.+?):(?<transport_type>.+?):(?<logical_sector_size>\d+):(?<physical_sector_size>\d+):(?<partition_table_type>.+?):(?<model_name>.+?):(?<flags>.*?);$/ =~ parted_lines[1]
+      @units = units
+      @size = size
+      @transport_type = transport_type
+      @logical_sector_size = logical_sector_size.to_i
+      @physical_sector_size = physical_sector_size.to_i
+      @partition_table_type = partition_table_type
+      @model_name = model_name
+      @flags = flags.length > 0 ? flags : nil
+    end
+  end
+end
+
+Facter.add(:disks_partition_table) do
+  confine kernel: "Linux"
+
+  confine :disks do |value|
+    value != nil
+  end
+
+  confine do
+    Process.euid === 0
```

Comment:
I wonder if it should use the `identity.uid` fact.

Reply:
Refactored to use `identity.uid` instead.